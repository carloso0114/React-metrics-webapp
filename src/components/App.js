import { useSelector, useDispatch } from 'react-redux';
import { useEffect } from 'react';
import {
  BrowserRouter as Router,
  Route,
  Switch,
} from 'react-router-dom';
import HomePage from './homePage/homePage';
import Details from './details.jsx/details';
import { fetchData } from '../redux/coronaData/coronaData.actions';
import NavBar from './navBar/navBar';

export default function App() {
  const { dataReducer } = useSelector((state) => state);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchData());
  }, []);
  return (
    <div>
      <Router>
        <Switch>
          <Route exact path="/">
            <NavBar />
            <HomePage data={dataReducer} />
          </Route>
          { dataReducer.map((data) => (
            <Route key={data[1].id} path={`/${data[0].toLowerCase()}`}>
              <Details data={data} />
            </Route>
          ))}
        </Switch>
      </Router>
    </div>
  );
}
