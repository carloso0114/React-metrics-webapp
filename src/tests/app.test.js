import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import App from '../components/App';
import store from '../redux/configureStore';

it('renders correctly', () => {
  const tree = renderer
    .create(<Router><Provider store={store}><App /></Provider></Router>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
