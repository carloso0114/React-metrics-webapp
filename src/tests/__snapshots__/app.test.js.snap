// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <nav
    className="sc-gKseQn hBAThJ navbar"
  >
    <a
      className="sc-iBPTik dNajSQ"
      href="/"
      onClick={[Function]}
    >
       &lt; Home
    </a>
    <span
      className="sc-jSgvzq gOpCnz"
    >
      COVID Analitycs Data
    </span>
    <span>
      <i
        className="bi bi-mic-fill h4 me-3"
      />
      <i
        className="bi bi-gear-fill h4"
      />
    </span>
  </nav>
  <div
    className="container-fluid"
  >
    <div
      className="sc-bdfBQB lSopt"
    >
      <label
        className="sc-gsTEea bplxbl"
      >
        Search 
      </label>
      <input
        className="sc-dlfnuX fVXMgl ms-2"
        name="search"
        onChange={[Function]}
        placeholder="Enter country name"
        type="text"
        value=""
      />
    </div>
    <div
      className="row"
    />
  </div>
</div>
`;
